### DISCUSSION TITLE
Old method of getting zoomed in minIndex and maxIndex of visible points no longer works.

### DISCUSSION THREAD

**Discussion by twilson90** [chartjs/chartjs-plugin-zoom#855] [General]
I have a zoomable line chart and I need to know the visible dataset indices.
According to old answers on stackoverflow `chart.scales.x.minIndex` and `chart.scales.x.maxIndex` were available in older versions, but they no longer appear to exist in the latest version of chartjs.
I can get the value range with `_range` but I would need to process the entire dataset to find the indices from that, and my dataset is rather massive (which is why I'm using zoom)

**Comment by gopal-panigrahi** [comment]
@twilson90 Can you provide the relevant code with which you are having issue .

**Comment by kurkle** [comment]
Chart.js uses binary search internally to find the visible indices, but does not store them.
See https://github.com/chartjs/Chart.js/blob/master/src/helpers/helpers.extras.ts#L87
